### PR TITLE
[DO NOT MERGE] Microphone streaming - multi-threaded

### PR DESCRIPTION
**Pull request by nirupa-kumar**
For @beccca preliminary review

### CODE REVIEW

File: speech/cloud-client/src/main/java/com/example/speech/MicStreamRecognize.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2018 Google Inc.
```
Comment (dzlier-gcp) [inline speech/cloud-client/src/main/java/com/example/speech/MicStreamRecognize.java]: s/Inc./LLC/

This applies to all files.

File: speech/cloud-client/src/main/java/com/example/speech/MicStreamRecognize.java [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+          RecognitionConfig.newBuilder()
+              .setEncoding(AudioEncoding.LINEAR16)
+              .setLanguageCode("en-US")
+              .setSampleRateHertz(16000)
+              .setModel("default")
+              .build();
+      StreamingRecognitionConfig config =
+          StreamingRecognitionConfig.newBuilder().setConfig(recConfig).build();
+
+      class ResponseApiStreamingObserver<T> implements ApiStreamObserver<T> {
```
Comment (dzlier-gcp) [inline speech/cloud-client/src/main/java/com/example/speech/MicStreamRecognize.java]: Unless I'm missing something, I don't see any need for this class to be defined during the execution of this method. Please move the definition outside of the method body (or provide comments explaining why it is defined here, if necessary).

**Comment by nnegrey** [comment]
Should we push this code into the Recognize.java file?

**Comment by nirupa-kumar** [comment]
Felt adding it in as part of the Recognize.java file made the most sense too. please do let me know which way is best.

File: speech/cloud-client/src/main/java/com/example/speech/MicStreamRecognize.java [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+import javax.sound.sampled.TargetDataLine;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.List;
+
+/**
+ * Google Cloud SpeechToText API sample application. Example usage: mvn package exec:java
+ * -Dexec.mainClass='com.example.speech.MicStreamRecognize' -Dexec.args="micstreamrecognize
+ * <duration>"
```
Comment (nnegrey) [inline speech/cloud-client/src/main/java/com/example/speech/MicStreamRecognize.java]: Can use the short version: `mvn exec:java -DMicStreamRecognize -Dexec.args="micstreamrecognize <duration>"

File: speech/cloud-client/src/test/java/com/example/speech/MicStreamRecognizeIT.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @After
+    public void tearDown() {
+        System.setOut(null);
+    }
+
+    @Test
+    public void testMicStreamRecognize() throws Exception {
+        Recognize.streamingRecognizeFile(audioFileName);
```
Comment (nnegrey) [inline speech/cloud-client/src/test/java/com/example/speech/MicStreamRecognizeIT.java]: Is this testing the above file or just the Recognize.java streaming?

**Comment by nirupa-kumar** [comment]
Will be closing this and re-opening a new one since there have been significant changes to the underlying code.

**Pull request closed without merging**